contracts/access/SignatureBouncer.sol

```diff
@@ -20,12 +20,23 @@ import "../ECRecovery.sol";
  * @dev Then restrict access to your crowdsale/whitelist/airdrop using the
  * @dev `onlyValidSignature` modifier (or implement your own using isValidSignature).
  * @dev
+ * @dev In addition to `onlyValidSignature`, `onlyValidSignatureAndMethod` and
+ * @dev `onlyValidSignatureAndData` can be used to restrict access to only a given method
+ * @dev or a given method with given parameters respectively.
+ * @dev
  * @dev See the tests Bouncer.test.js for specific usage examples.
+ * @notice A method that uses the `onlyValidSignatureAndData` modifier must make the _sig
+ * @notice parameter the "last" parameter. You cannot sign a message that has its own
+ * @notice signature in it so the last 160 bytes is of msg.data (which represents _sig)
```

Comment:
Can we update this to the new value of 128 and also mention that for this to be easy to construct signatures for, your arguments must be fixed-sized?

Reply:
(sorry, just took one last pass through it and I think it's worth adding a comment about this)

Reply:
@shrugs  Ah, sorry, my bad. I'll get that changed. I actually went and posted a detailed answer on a stack overflow thread about how `msg.data` is constructed https://ethereum.stackexchange.com/a/50616/40921 . Do you want me to link this thread in the comments?

Reply:
sure, that's a solid answer